crypto/ctype.c

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <e_os.h>
```

Comment:
I must be confused. I have recollection that it was resolved in previous version. I mean did complain about it earlier, and as far as I recall you've fixed it... But anyway...

Before I say something about moving OSSL_NELEM, since it came to this, I feel compelled to say that I for one customarily speak against project-specific lingo for trivial constructs [such as this one]. I find it appropriate in cases when it saves on-screen space and/or appears in tight repetition pattern. Or in other words when it pays off keeping lingo in mind. For example in this case I see no point in using OSSL_NELEM. [Do note that I didn't say that I see no point in OSSL_NELEM. I mean even though I would probably be against using it in most cases, it's not like I don't recognize its value in more complex cases. As for "would be against". I normally restrain myself from speaking up, unless it comes up as explicit point in discussion, like here.] Back to anyway...

e_os.h is a "heavy-weight" header that is supposed to cover real OS specifics. Such that are not covered by C language specification, or POSIX, or SUS... Less reliance on it is better(\*). Things that are platform-neutral, such as OSSL_NELEM, do belong elsewhere. Since I don't really see compelling need for it, I don't have strong opinion on where it should go. Perhaps include/internal/cryptlib.h would be appropriate?

(\*) I wish there was no need for it at all. Indeed, wouldn't it be nice if whole code-base could be platform-neutral :-)